### ISSUE TITLE
Problem with sniff `WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents`?

### ISSUE THREAD

**Issue by leoloso** [10up/plugin-check#266]
### Is your enhancement related to a problem? Please describe.

I'm getting the following error when checking my plugin:

```
WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents Line 195 of file src/ContentProcessors/AbstractContentParser.php.
file_get_contents() is discouraged. Use wp_remote_get() for remote URLs instead.

$fileContent = file_get_contents($file);
```

But I'm using `file_get_contents` to read the contents of a local file (passing a path to the local file, not an URL), so the suggestion to instead use `wp_remote_get` seems to be wrong.

Maybe this sniff is wrong and should not be enforced?

This is the code:

```php
public function getContent(
  string $filename,
  string $extension,
  string $relativePathDir = '',
  array $options = []
): string {
  // Make sure the relative path ends with "/"
  if ($relativePathDir) {
    $relativePathDir = \trailingslashit($relativePathDir);
  }
  $localeLanguage = $this->getLocaleHelper()->getLocaleLanguage();
  $localizeFile = \trailingslashit($this->getFileDir()) . $relativePathDir . $filename . '/' . $localeLanguage . '.' . $extension;
  if (file_exists($localizeFile)) {
    // First check if the localized version exists
    $file = $localizeFile;
  } else {
    // Otherwise, use the default language version
    $defaultDocsLanguage = $this->getDefaultDocsLanguage();
    $file = \trailingslashit($this->getFileDir()) . $relativePathDir . $filename . '/' . $defaultDocsLanguage . '.' . $extension;
    // Make sure this file exists
    if (!file_exists($file)) {
      throw new ContentNotExistsException(sprintf(
        \__('File \'%s\' does not exist', 'gatographql'),
        $file
      ));
    }
  }
  $fileContent = file_get_contents($file);
  if ($fileContent === false) {
    throw new ContentNotExistsException(sprintf(
      \__('File \'%s\' is corrupted', 'gatographql'),
      $file
    ));
  }
  $htmlContent = $this->getHTMLContent($fileContent);
  $pathURL = \trailingslashit($this->baseURL . $this->docsFolder) . $relativePathDir . $filename . '/';
  // Include the images from the GitHub repo, unless we are in DEV
  if (!RootEnvironment::isApplicationEnvironmentDev()) {
    $options[self::PATH_URL_TO_DOCS] = \trailingslashit($this->githubRepoDocsRootPathURL . $this->docsFolder) . $relativePathDir . $filename . '/';
  }
  return $this->processHTMLContent($htmlContent, $pathURL, $options);
}
```

### Designs

<img width="992" alt="Screenshot 2023-09-22 at 11 17 43 AM" src="https://github.com/10up/plugin-check/assets/1981996/d056d20d-e630-4d53-acea-f5d1a15ebb55">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leoloso** [comment]
Btw, I've seen that this sniff is ignored in the plugin: https://github.com/10up/plugin-check/blob/42c2b20e41ff3fd0f2874d286c4af9872e6ff5f4/includes/Checker/Checks/Abstract_File_Check.php#L181

**Comment by swissspidy** [comment]
This is similar to #430

The sniff is not super smart and most importantly, it does not _run_ code. It does not know what `$file` is. That's why it errs on the side of caution and tells you that it's a potentially discouraged function. That's also why it's a _warning_ and not an _error_.

Ignoring the sniff is the right approach in such a case.

**Comment by davidperezgar** [comment]
I think this issue should be in WordPress Coding Standards.

**Comment by swissspidy** [comment]
Even there, there is nothing that can be done. The sniff doesn‘t know whether it‘s a file or not. This is really a wontfix.
